tests/unit-core/test-container-operators.c

```diff
@@ -0,0 +1,118 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+#include "test-common.h"
+
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  if (!jerry_is_feature_enabled (JERRY_FEATURE_MAP)
+      || !jerry_is_feature_enabled (JERRY_FEATURE_SET)
+      || !jerry_is_feature_enabled (JERRY_FEATURE_WEAKMAP)
+      || !jerry_is_feature_enabled (JERRY_FEATURE_WEAKSET))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Containers are disabled!\n");
+    jerry_cleanup ();
+    return 0;
+  }
+
+  jerry_value_t map = jerry_create_container (JERRY_CONTAINER_TYPE_MAP, NULL, 0);
+  TEST_ASSERT (jerry_get_container_type (map) == JERRY_CONTAINER_TYPE_MAP);
+
+  jerry_value_t key_str = jerry_create_string ((jerry_char_t *) "number");
+  jerry_value_t number = jerry_create_number (10);
+  jerry_value_t args[2] = {key_str, number};
+  jerry_value_t value = jerry_container_operation (JERRY_CONTAINER_OP_SET, map, args, 2);
+  TEST_ASSERT (!jerry_value_is_error (value));
+  jerry_release_value (value);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_GET, map, &key_str, 1);
+  TEST_ASSERT (jerry_get_number_value (value) == 10);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_HAS, map, &key_str, 1);
```

Comment:
Please group your tests somehow to avoid assigning multiple values in a big method for a single variable.
(for inspiration you can check the `test-api-promise.c` file)